### PR TITLE
[AUTO-MERGE] Merge main into develop via merge-main-to-develop

### PR DESCRIPTION
**Pull request by idanov**
A new change in main cannot be merged into develop as part of the regular sync job, hence this PR. Please resolve the conflicts manually, and make sure to obtain 2 approvals once the builds pass.

### IMPORTANT NOTICE

Please let CircleCI merge this PR automatically, with merge commit enabled.

<a href="https://gitpod.io/#https://github.com/kedro-org/kedro/pull/1263"></a>

### CODE REVIEW

**Review by MerelTheisenQB — APPROVED** [review]
LGTM! 👍 

It's weird that the coverage is failing for `tests/framework/cli/micropkg/conftest.py `.. Firstly, I'm wondering if coverage should be run on test files at all (I raised this point in my PR for python 3.10 as well), but secondly, this conftest was copied right, so how come it was covered fine before? 🤔

**Comment by SajidAlamQB** [comment]
> LGTM! 👍
> 
> It's weird that the coverage is failing for `tests/framework/cli/micropkg/conftest.py `.. Firstly, I'm wondering if coverage should be run on test files at all (I raised this point in my PR for python 3.10 as well), but secondly, this conftest was copied right, so how come it was covered fine before? 🤔

I think it's because those lines were covered in `test_pipeline.py` an equivalent to this test file was not made for micropkg as it wasn't needed.

**Comment by MerelTheisenQB** [comment]
> I think it's because those lines were covered in `test_pipeline.py` an equivalent to this test file was not made for micropkg as it wasn't needed. More specifically when we ran the test `test_catalog_and_params`.

Ah okay, so maybe the fixture can be cleaned up a bit?

File: kedro/framework/cli/pipeline.py [hunk 1]
```diff
@@ -239,6 +239,11 @@ def pull_package(  # pylint:disable=unused-argument, too-many-arguments
     **kwargs,
 ) -> None:
     """Pull and unpack a modular pipeline in your project."""
+    deprecation_message = (
+        "DeprecationWarning: Command `kedro pipeline pull` is deprecated in Kedro 0.18.0. "
+        "Please use `kedro micropkg pull` instead."
+    )
+    click.secho(deprecation_message, fg="red")
```
Comment (lorenabalan) [inline kedro/framework/cli/pipeline.py:246]: Let's not forget to remove these commands on `develop`! 🙌

File: tests/framework/cli/micropkg/test_micropkg_package.py [hunk 2]
```diff
@@ -0,0 +1,581 @@
...
+        self.assert_sdist_contents_correct(
+            sdist_location=sdist_location, package_name=package_name, version="0.1"
+        )
+
+    def test_micropkg_package_same_name_as_package_name(
+        self, fake_metadata, fake_project_cli, fake_repo_path
+    ):
+        """Create modular pipeline with the same name as the
+        package name, then package as is. The command should run
+        and the resulting wheel should have all expected contents.
```
Comment (lorenabalan) [inline tests/framework/cli/micropkg/test_micropkg_package.py:87]:
```suggestion
        and the resulting sdist should have all expected contents.
```

File: tests/framework/cli/micropkg/test_micropkg_package.py [hunk 3]
```diff
@@ -0,0 +1,581 @@
...
+        self.assert_sdist_contents_correct(
+            sdist_location=sdist_location, package_name=pipeline_name
+        )
+
+    def test_micropkg_package_same_name_as_package_name_alias(
+        self, fake_metadata, fake_project_cli, fake_repo_path
+    ):
+        """Create modular pipeline, then package under alias
+        the same name as the package name. The command should run
+        and the resulting wheel should have all expected contents.
```
Comment (lorenabalan) [inline tests/framework/cli/micropkg/test_micropkg_package.py:113]:
```suggestion
        and the resulting sdist should have all expected contents.
```

File: tests/framework/cli/micropkg/test_micropkg_package.py [hunk 4]
```diff
@@ -0,0 +1,581 @@
...
+        assert result.exit_code == 0
+        success_message = (
+            f"`dummy_package.pipelines.{PIPELINE_NAME}` packaged! "
+            f"Location: {destination}"
+        )
+        assert success_message in result.output
+
+        self.assert_sdist_contents_correct(sdist_location=destination)
+
+    def test_micropkg_package_overwrites_wheel(
```
Comment (lorenabalan) [inline tests/framework/cli/micropkg/test_micropkg_package.py:167]:
```suggestion
    def test_micropkg_package_overwrites_sdist(
```

**Review by lorenabalan — COMMENTED** [review]
quick comments before final review (got 2 files left)

File: kedro/framework/cli/pipeline.py (hunk 1 above)
Comment (SajidAlamQB) [reply kedro/framework/cli/pipeline.py:246]: Will create a second PR to refactor these on `develop` once this lands.

**Review by lorenabalan — APPROVED** [review]
LGTM!

File: kedro/framework/cli/pipeline.py (hunk 1 above)
Comment (lorenabalan) [reply kedro/framework/cli/pipeline.py:246]: Yeah that's what I meant, definitely not here.

**Pull request merged**